app/presenters/schools/courses/curriculum_presenter.rb

```diff
@@ -66,16 +67,24 @@ def targets
             id: target.id,
             target_group_id: target.target_group_id,
             title: target.title,
-            sortIndex: target.sort_index,
+            sort_index: target.sort_index,
             visibility: target.visibility
           }
         end
       end
 
       def vimeo_access_token?
-        token = @course.school.configuration['vimeo'] && @course.school.configuration['vimeo']['access_token'] ||
-          Rails.application.secrets.vimeo_access_token
-        token.present?
+        @vimeo_access_token ||= begin
+          token = @course.school.configuration['vimeo'] && @course.school.configuration['vimeo']['access_token'] ||
+            Rails.application.secrets.vimeo_access_token
+          token.present?
+        end
```

Comment:
Memoization of _boolean_ returns like this has a problem. If the returned value is `false`, the block will get executed again, because it fails the `||=` check.

One way to get around this is by using something like `instance_variable_defined?`

```suggestion
        return @vimeo_access_token if instance_variable_defined?(:@vimeo_access_token)
        @vimeo_access_token = @course.school.configuration.dig('vimeo', 'access_token').present? || Rails.application.secrets.vimeo_access_token.present?
```

No `begin-end` block required.